{
  "uid" : "219967405852c388",
  "name" : "TestBatchModifyFlies.test_batch_modify_files[pause_offer_schedule]",
  "title" : "Test batch modify files [pause_offer_schedule]",
  "time" : {
    "start" : 1560319751531,
    "stop" : 1560319760725,
    "duration" : 9194
  },
  "summary" : {
    "attachments" : 4,
    "steps" : 0
  },
  "failure" : {
    "message" : "Exception: 用例关联文件不存在\n文件路径： pause_offer.json",
    "stackTrace" : "test_name = 'pause_offer_schedule', param = 'pause_offer.json'\n_path = '/Users/wangjuan/workpace/Aff_service/aff/pages/offer', relevance = None\n\n    def read_param(test_name, param, _path, relevance=None):\n        \"\"\"\n        读取用例中参数parameter\n        :param test_name: 用例名称\n        :param param: parameter\n        :param relevance: 关联对象\n        :param _path: case路径\n        :param result: 全局结果\n        :return:\n        \"\"\"\n    \n        if isinstance(param, dict):\n            param = replace(param, relevance)\n        elif isinstance(param, list):\n            param = replace(param, relevance)\n        else:\n            try:\n>               with open(_path + \"/\" + param, \"r\", encoding=\"utf-8\") as f:\nE               FileNotFoundError: [Errno 2] No such file or directory: '/Users/wangjuan/workpace/Aff_service/aff/pages/offer/pause_offer.json'\n\nbin/unit/readParameter.py:30: FileNotFoundError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_postBatchModifyFlies.TestBatchModifyFlies object at 0x112be5358>\ncase_data = {'address': '/admin/offer/postBatchModifyFlies', 'check': [{'check_type': 'entirely_check', 'expected_code': 200, 'expected_request': 'result_pause_offer.json'}], 'cookies': True, 'file': True, ...}\n\n    @pytest.mark.parametrize(\"case_data\", case_dict[\"test_case\"],\n                             ids=[\"pause_offer_error\", \"pause_offer_schedule\"])\n    @allure.story(\"batch_modify_offer\")\n    @pytest.mark.flaky(reruns=3, reruns_delay=3)\n    def test_batch_modify_files(self, case_data):\n        \"\"\"\n    \n        :param case_data: 测试用例\n        :return:\n        \"\"\"\n        self.init_relevance = ini_request(case_dict, PATH)\n        # 发送测试请求\n>       api_send_check(case_data, case_dict, self.init_relevance, PATH)\n\naff/testcase/test_postBatchModifyFlies.py:34: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nbin/unit/apiSendCheck.py:23: in api_send_check\n    project_dict[\"test_info\"].get(\"address\"), _path, relevance)\nbin/unit/apiSend.py:31: in send_request\n    parameter = readParameter.read_param(data[\"test_name\"], data[\"parameter\"], _path, relevance)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ntest_name = 'pause_offer_schedule', param = 'pause_offer.json'\n_path = '/Users/wangjuan/workpace/Aff_service/aff/pages/offer', relevance = None\n\n    def read_param(test_name, param, _path, relevance=None):\n        \"\"\"\n        读取用例中参数parameter\n        :param test_name: 用例名称\n        :param param: parameter\n        :param relevance: 关联对象\n        :param _path: case路径\n        :param result: 全局结果\n        :return:\n        \"\"\"\n    \n        if isinstance(param, dict):\n            param = replace(param, relevance)\n        elif isinstance(param, list):\n            param = replace(param, relevance)\n        else:\n            try:\n                with open(_path + \"/\" + param, \"r\", encoding=\"utf-8\") as f:\n                    data = json.load(f)\n                    for i in data:\n                        if i[\"test_name\"] == test_name:\n                            param = i[\"parameter\"]\n                            break\n                    if not isinstance(param, dict):\n                        raise Exception(\"未能找到用例关联的参数\\n文件路径：%s\\n索引：%s\" % (param, _path))\n                    else:\n                        param = replace(param, relevance)\n            except FileNotFoundError:\n>               raise Exception(\"用例关联文件不存在\\n文件路径： %s\" % param)\nE               Exception: 用例关联文件不存在\nE               文件路径： pause_offer.json\n\nbin/unit/readParameter.py:41: Exception"
  },
  "description" : {
    "value" : "\n\n        :param case_data: 测试用例\n        :return:\n        ",
    "type" : "TEXT"
  },
  "severity" : "NORMAL",
  "status" : "FAILED",
  "testId" : null,
  "suite" : {
    "uid" : "f0ed89dd652314b4",
    "name" : "test_postBatchModifyFlies",
    "title" : "Test post batch modify flies"
  },
  "steps" : [ ],
  "attachments" : [ {
    "title" : "Captured log call",
    "source" : "9fe90090-5e88-4b11-8333-8399aad384cb-attachment.txt",
    "type" : "text/plain",
    "size" : 348,
    "uid" : "f4385fd30e753a3e"
  }, {
    "title" : "Captured log call",
    "source" : "954a1835-22a2-4b13-b5d3-77d8d6ea7e4e-attachment.txt",
    "type" : "text/plain",
    "size" : 348,
    "uid" : "f8f463d3f4ab0e88"
  }, {
    "title" : "Captured log call",
    "source" : "84017d67-130a-4a82-82dc-71685ff1c0d8-attachment.txt",
    "type" : "text/plain",
    "size" : 348,
    "uid" : "1091a61d1ca6df3b"
  }, {
    "title" : "Captured log call",
    "source" : "7234f14e-a7ba-426a-97b0-76f246e20f05-attachment.txt",
    "type" : "text/plain",
    "size" : 348,
    "uid" : "22d46db696015ebe"
  } ],
  "issues" : [ ],
  "labels" : [ {
    "name" : "thread",
    "value" : "41942-MainThread"
  }, {
    "name" : "suite-name",
    "value" : "test_postBatchModifyFlies"
  }, {
    "name" : "feature",
    "value" : "offer"
  }, {
    "name" : "story",
    "value" : "batch_modify_offer"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "suite-title",
    "value" : null
  }, {
    "name" : "severity",
    "value" : "normal"
  }, {
    "name" : "host",
    "value" : "xiaoxideMacBook-Pro.local"
  }, {
    "name" : "language",
    "value" : "cpython3"
  } ],
  "parameters" : [ ]
}